12.0.Dockerfile

```diff
@@ -0,0 +1,165 @@
+    PATH="/home/odoo/.local/bin:$PATH" \
+    PIP_NO_CACHE_DIR=0 \
+    PTVSD_ARGS="--server-host 0.0.0.0 --port 6899" \
+    PTVSD_ENABLE=0 \
+    PUDB_RDB_HOST=0.0.0.0 \
+    PUDB_RDB_PORT=6899 \
+    PYTHONOPTIMIZE=1 \
+    UNACCENT=true \
+    WAIT_DB=true \
+    WDB_NO_BROWSER_AUTO_OPEN=True \
+    WDB_SOCKET_SERVER=wdb \
+    WDB_WEB_PORT=1984 \
+    WDB_WEB_SERVER=localhost
+
+# Other requirements and recommendations to run Odoo
+# See https://github.com/$ODOO_SOURCE/blob/$ODOO_VERSION/debian/control
+RUN apt-get -qq update \
+    && apt-get -yqq upgrade \
+    && apt-get install -yqq --no-install-recommends \
+        chromium \
+        gettext-base \
+        gnupg2 \
+        locales-all \
+        nano \
+        telnet \
+        vim \
+        zlibc \
+    && echo 'deb http://apt.postgresql.org/pub/repos/apt/ stretch-pgdg main' >> /etc/apt/sources.list.d/postgresql.list \
+    && curl -SL https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - \
+    && curl https://bootstrap.pypa.io/get-pip.py | python3 /dev/stdin \
+    && curl -sL https://deb.nodesource.com/setup_8.x | bash - \
+    && apt-get update \
+    && apt-get install -yqq --no-install-recommends nodejs postgresql-client \
+    && curl -SLo wkhtmltox.deb https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/${WKHTMLTOPDF_VERSION}/wkhtmltox_${WKHTMLTOPDF_VERSION}-1.stretch_amd64.deb \
+    && echo "${WKHTMLTOPDF_CHECKSUM}  wkhtmltox.deb" | sha256sum -c - \
+    && apt-get install -yqq --no-install-recommends ./wkhtmltox.deb \
+    && rm wkhtmltox.deb \
+    && wkhtmltopdf --version \
+    && rm -Rf /var/lib/apt/lists/* /tmp/*
+
+# Special case to get latest Less
+RUN ln -s /usr/bin/nodejs /usr/local/bin/node \
+    && npm install -g less \
+    && rm -Rf ~/.npm /tmp/*
+
+# Other facilities
+WORKDIR /opt/odoo
+RUN pip install \
+    astor git-aggregator openupgradelib ptvsd pudb wdb
+COPY bin/* /usr/local/bin/
+COPY lib/odoobaselib /usr/local/lib/python3.7/dist-packages/odoobaselib
+COPY build.d common/build.d
+COPY conf.d common/conf.d
+COPY entrypoint.d common/entrypoint.d
+RUN mkdir -p auto/addons custom/src/private \
+    && ln /usr/local/bin/direxec common/entrypoint \
+    && ln /usr/local/bin/direxec common/build \
+    && chmod -R a+rx common/entrypoint* common/build* /usr/local/bin \
+    && chmod -R a+rX /usr/local/lib/python3.7/dist-packages/odoobaselib \
+    && sync
+
+# Execute installation script by Odoo version
+# This is at the end to benefit from cache at build time
+# https://docs.docker.com/engine/reference/builder/#/impact-on-build-caching
+ARG ODOO_SOURCE=OCA/OCB
+ARG ODOO_VERSION=12.0
+ENV ODOO_VERSION="$ODOO_VERSION"
+RUN pip install -r https://raw.githubusercontent.com/odoo/odoo/saas-11.5/requirements.txt \
+    && pip install pg_activity \
+    && (python3 -m compileall -q /usr/local/lib/python3.7/ || true)
+
+# HACK Special case for Werkzeug
+USER odoo
+RUN pip install --user Werkzeug==0.14.1
```

Comment:
Why freezing this version? Use the one declared in Odoo requirements.txt

Reply:
See https://github.com/Tecnativa/doodba/issues/19